Automattic/wp-calypso
# A4A: Implement the Pressable section's plan selector.

This PR implements the Pressable plan selector in the new hosting page.


**Normal mode**
<img width="1532" alt="Screenshot 2024-12-13 at 10 50 55 PM" src="https://github.com/user-attachments/assets/ff549286-94ff-4fe3-b2f0-a6d52b676df5" />

**Referral Mode**
<img width="1557" alt="Screenshot 2024-12-13 at 10 48 16 PM" src="https://github.com/user-attachments/assets/5db9b8f7-7acf-45ae-beeb-d996d70408b0" />


**Regular ownership**
<img width="1601" alt="Screenshot 2024-12-13 at 10 51 21 PM" src="https://github.com/user-attachments/assets/369c5b38-dc3e-4dd9-afe7-c2010198e270" />



Depends on https://github.com/Automattic/wp-calypso/pull/97411
Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1550


## Proposed Changes

* Enhance the Filter component by adding a new 'Storage' view type. This change will impact production, but it’s acceptable since it maintains all existing functionalities while introducing additional value. Additionally, It has a new property to enable High-resource sites option. Note that the option is only available on the new page.
* Implement the Plan selector section based on the new design.

## Why are these changes being made?

* This is part of the In-product hosting page project.

## Testing Instructions

* Use the A4A live link and go to the `/marketplace/hosting/pressable` page.
* Verify that the plan selector is working as expected similar to how the production works with the addition of the High-resource sites option.

## Pre-merge Checklist
